### PR TITLE
Move Content-related models into a subdirectory

### PR DESCRIPTION
**Pull request by alecgibson**
This change moves the following classes from `models` into a
`models/content` subdirectory, with a corresponding `Content` module:

* `ContentItem`
* `Group`
* `Link`

### CODE REVIEW

**Comment by alecgibson** [comment]
[Move Content Models into a subdirectory](https://trello.com/c/tjGBH538/417-move-content-models-into-a-subdirectory)

File: app/models/content/content_item.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
+module Content
+  class ContentItem < ApplicationRecord
```
Comment (andrewgarner) [inline app/models/content/content_item.rb]: @pmanrubia - what are your thoughts on `Content::Item`?

Comment (alecgibson) [inline app/models/content/content_item.rb]: I'd considered doing that. I don't mind either way. I guess it just gets a bit weird if you refer within the module to just `Item`s

Comment (andrewgarner) [inline app/models/content/content_item.rb]: I think that is okay. There is always the option of fully qualifying it with `::Content::Item` where just `Item` might be confusing.

**Comment by alecgibson** [comment]
@andrewgarner they're showing as additions/deletions because the whole file has been nested inside a `module` - I don't know if there's any way to get around that without doing it in two (unatomic) commits?

File: app/models/content/content_item.rb (hunk 1 above)
Comment (alecgibson) [inline app/models/content/content_item.rb]: @andrewgarner done.

**Comment by andrewgarner** [comment]
@alecgibson yep. I'm just reading through https://cirw.in/blog/constant-lookup.html for ideas. I wonder if maintaining git history is better that an atomic commit in this instance?

File: app/models/content/content_item.rb (hunk 1 above)
Comment (pmanrubia) [inline app/models/content/content_item.rb]: 👍 to Content::Item

**Pull request closed without merging**